osu.Game.Tests/Visual/RealtimeMultiplayer/TestSceneRealtimeReadyButton.cs

```diff
@@ -0,0 +1,129 @@
+{
+    public class TestSceneRealtimeReadyButton : RealtimeMultiplayerTestScene
+    {
+        private RealtimeReadyButton button;
+
+        private BeatmapManager beatmaps;
+        private RulesetStore rulesets;
+
+        [BackgroundDependencyLoader]
+        private void load(GameHost host, AudioManager audio)
+        {
+            Dependencies.Cache(rulesets = new RulesetStore(ContextFactory));
+            Dependencies.Cache(beatmaps = new BeatmapManager(LocalStorage, ContextFactory, rulesets, null, audio, host, Beatmap.Default));
+            beatmaps.Import(TestResources.GetTestBeatmapForImport(true)).Wait();
+        }
+
+        [SetUp]
+        public new void Setup() => Schedule(() =>
+        {
+            var beatmap = beatmaps.GetAllUsableBeatmapSetsEnumerable(IncludedDetails.All).First().Beatmaps.First();
+
+            Beatmap.Value = beatmaps.GetWorkingBeatmap(beatmap);
+
+            Child = button = new RealtimeReadyButton
+            {
+                Anchor = Anchor.Centre,
+                Origin = Anchor.Centre,
+                Size = new Vector2(200, 50),
+                SelectedItem =
+                {
+                    Value = new PlaylistItem
+                    {
+                        Beatmap = { Value = beatmap },
+                        Ruleset = { Value = beatmap.Ruleset }
+                    }
+                }
+            };
+
+            Client.AddUser(API.LocalUser.Value);
+        });
+
+        [Test]
+        public void TestToggleStateWhenNotHost()
+        {
+            AddStep("add second user as host", () =>
+            {
+                Client.AddUser(new User { Id = 2, Username = "Another user" });
+                Client.TransferHost(2);
+            });
+
+            addClickButtonStep();
+            AddAssert("user is ready", () => Client.Room?.Users[0].State == MultiplayerUserState.Ready);
+
+            addClickButtonStep();
+            AddAssert("user is idle", () => Client.Room?.Users[0].State == MultiplayerUserState.Idle);
+        }
+
+        [TestCase(true)]
+        [TestCase(false)]
+        public void TestToggleStateWhenHost(bool allReady)
+        {
+            AddStep("setup", () =>
+            {
+                Client.TransferHost(Client.Room?.Users[0].UserID ?? 0);
+
+                if (!allReady)
+                    Client.AddUser(new User { Id = 2, Username = "Another user" });
+            });
+
+            addClickButtonStep();
+            AddAssert("user is ready", () => Client.Room?.Users[0].State == MultiplayerUserState.Ready);
+
+            addClickButtonStep();
+            AddAssert("match started", () => Client.Room?.Users[0].State == MultiplayerUserState.WaitingForLoad);
+        }
+
+        [Test]
+        public void TestBecomeHostWhileReady()
+        {
+            addClickButtonStep();
+            AddStep("make user host", () => Client.TransferHost(Client.Room?.Users[0].UserID ?? 0));
```

Comment:
There's actually no second user in this room, so this step has no effect and this test doesn't test what it says it does. The current user already has host.